core/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/FilterAllocationDecider.java

```diff
@@ -96,7 +99,8 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
             if (initialRecoveryFilters != null  &&
                 RecoverySource.isInitialRecovery(shardRouting.recoverySource().getType()) &&
                 initialRecoveryFilters.match(node.node()) == false) {
-                return allocation.decision(Decision.NO, NAME, "node does not match index initial recovery filters [%s]",
+                return allocation.decision(Decision.NO, NAME,
+                    "initial allocation of the shrunken index is only allowed on nodes [%s] that hold a copy of every shard in the index",
```

Comment:
I was on the fence about this too.  The initial recovery filter setting is an internal setting only set by the shrink API.  Since the regular user does not access that setting, it seemed strange to me to give a general message about initial recovery filters that they would not know anything about.  That's why I went with this message.  WDYT?